### PR TITLE
[pvr] only use provider icon as fallback for recording if no addon supplied icon is available

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

Only use the provider icon as fallback for a PVR recording if no add-on supplied icon is available.

## Motivation and context

If an add-on supplies an icon for a recording it should always be used before the fallback.

## How has this been tested?

On OSX Monterrey with pvr.vuplus.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/pvr/providers/PVRProviders.h [hunk 1]
```diff
@@ -43,6 +43,12 @@ class CPVRProvidersContainer
      */
   std::vector<std::shared_ptr<CPVRProvider>> GetProvidersList() const;
 
+  /*!
+     * Get the number of providers in this container
+     * @return The total number of providers
+     */
+  int GetNumProviders() const;
```
Comment (ksooo) [inline xbmc/pvr/providers/PVRProviders.h:50]: Signed ? Can there be less than 0 providers? ;-)

Comment (phunkyfish) [reply xbmc/pvr/providers/PVRProviders.h:50]: Whoops, fixed :)

File: xbmc/pvr/guilib/guiinfo/PVRGUIInfo.cpp [hunk 2]
```diff
@@ -478,7 +479,10 @@ bool CPVRGUIInfo::GetListItemAndPlayerLabel(const CFileItem* item, const CGUIInf
         break;
       }
       case LISTITEM_ICON:
-        if (!recording->Channel())
+        if (recording->ClientIconPath().empty() &&
+            recording->ClientThumbnailPath().empty() &&
+            !recording->Channel() &&
+            CServiceBroker::GetPVRManager().Providers()->GetNumProviders() > 1)
```
Comment (ksooo) [inline xbmc/pvr/guilib/guiinfo/PVRGUIInfo.cpp:485]: 1. Could you explain why > 1, not > 0? Maybe this is even worth a comment in the code?
2. Looking at the implementation of `CPVRRecording::Channel()`, this seems to be a quite expensive operation. I suggest to add the check as last term of the if clause.

**Review by ksooo — APPROVED** [review]
LGTM.

**Pull request merged**